biahub/track.py

```diff
@@ -0,0 +1,993 @@
+    ... )
+    """
+
+    output_dirpath = Path(output_dirpath)
+
+    settings = yaml_to_model(config_filepath, TrackingSettings)
+    tracking_cfg = settings.tracking_config
+
+    # Get the shape of the data
+    with open_ome_zarr(input_position_dirpaths[0]) as dataset:
+        T, C, Z, Y, X = dataset.data.shape
+        scale = dataset.scale
+        shape = (T, C, Z, Y, X)
+
+    # Resolve z-slices
+    z_slices = resolve_z_slice(settings.z_slices, shape[2], settings.mode)
+
+    # Define output metadata
+    if settings.mode == "2D":
+        output_shape = (T, 1, 1, Y, X)
+        track_scale = scale[-2:]
+    else:
+        output_shape = (T, 1, Z, Y, X)
+        track_scale = scale[-3:]
+
+    output_metadata = {
+        "shape": output_shape,
+        "chunks": None,
+        "scale": scale,
+        "channel_names": [f"{settings.target_channel}_labels"],
+        "dtype": np.uint32,
+    }
+
+    # Create the output zarr mirroring input_position_dirpaths
+    create_empty_hcs_zarr(
+        store_path=output_dirpath,
+        position_keys=[p.parts[-3:] for p in input_position_dirpaths],
+        **output_metadata,
+    )
+
+    # Estimate resources
+    num_cpus, gb_ram_per_cpu = estimate_resources(
+        shape=[T, C, Z, Y, X], ram_multiplier=16, max_num_cpus=16
+    )
+
+    # Use the number of CPUs and RAM per CPU in the tracking configuration
+    tracking_cfg["segmentation_config"]["n_workers"] = num_cpus
+    tracking_cfg["linking_config"]["n_workers"] = num_cpus
+
+    # Create default instance
+    default_config = MainConfig()
+    tracking_cfg = update_model(default_config, tracking_cfg)
+
+    # Prepare SLURM arguments
+    slurm_args = {
+        "slurm_job_name": "tracking",
+        "slurm_mem_per_cpu": f"{gb_ram_per_cpu}G",
+        "slurm_cpus_per_task": num_cpus,
+        "slurm_array_parallelism": 100,  # process up to 100 positions at a time
+        "slurm_time": 60,
+        "slurm_partition": "preempted",
+        "slurm_num_gpus": 1,
+        "slurm_use_srun": False,
+    }
+
+    # Override defaults if sbatch_filepath is provided
+    if sbatch_filepath:
+        slurm_args.update(sbatch_to_submitit(sbatch_filepath))
+
+    # Run locally or submit to SLURM
+    if local:
+        cluster = "local"
+    else:
+        cluster = "slurm"
+
+    # Prepare and submit jobs
+    click.echo(f"Preparing jobs: {slurm_args}")
+    slurm_out_path = output_dirpath.parent / "slurm_output"
+    executor = submitit.AutoExecutor(folder=slurm_out_path, cluster=cluster)
+    executor.update_parameters(**slurm_args)
+
+    click.echo('Submitting SLURM jobs...')
+    jobs = []
+
+    with executor.batch():
+        for input_position_path in input_position_dirpaths:
+            position_key = input_position_path.parts[-3:]
+            job = executor.submit(
+                track_one_position,
+                position_key=position_key,
+                output_dirpath=output_dirpath,
```

Comment:
Here you should iterate over iohub Position objects like we do in all other CLIs. This way you won't need to pass the path to the data using the tracking config file, related to this other comment: https://github.com/czbiohub-sf/biahub/pull/93#discussion_r2162570350